saleae/saleae.py

```diff
@@ -379,15 +382,20 @@ def get_active_channels(self):
 		return digital, analog
 
 	def set_active_channels(self, digital, analog):
+		digital_no = 0 if digital is None else len(digital)
+		analog_no = 0 if analog is None else len(analog)
+		if digital_no <= 0 and analog_no <= 0:
+			raise ValueError('bad active channels, has to set at least one active channel')
```

Comment:
Elsewhere, I raise a ImpossibleSettings (derived from Saleae.SaleaeError) for things like this. I think that's slightly more helpful than a generic ValueError (although it may may sense to mix-in ValueError to the ImpossibleSettings class).